FlatTensor alignment tests

lucylq opened the pull request:
Summary:
1. Pass segment_alignment from ExecutorchBackendConfig to flat tensor serializer
2. Set segment_alignment=128 as default to match ExecutorchBackendConfig
3. Remove tensor_alignment from the config (do not have multiple tensors per segment)
4. Additional tests for varying segment alignment.

Differential Revision: D89422691

larryliu0820 reviewed (approved):
Review automatically exported from Phabricator review in Meta.

lucylq committed:
FlatTensor alignment tests (#16317)

Summary:

1. Pass segment_alignment from ExecutorchBackendConfig to flat tensor serializer

Other changes:
2. Set segment_alignment=128 as default to match ExecutorchBackendConfig
3. Remove tensor_alignment from the config (we do not have multiple tensors per segment anymore)
4. Set flatbuffer alignment=16. I don't think we need any alignment at the moment here, but can't hurt. The PTE file aligns flatbuffer to the largest force_align value in the schema, which the PTD file doesn't have.
5. Additional tests for varying segment alignment.

Reviewed By: larryliu0820

Differential Revision: D89422691
